client/valor/coretypes.py

```diff
@@ -1458,6 +1471,7 @@ def create_predictions(
         dataset: Dataset,
         model: Model,
         predictions: List[Prediction],
+        timeout: Optional[float] = None,
```

Comment:
which functions do you expect to have retry logic after this PR, and which functions won't have retry logic? it's a little hard to follow here since the default is two seconds in `_requests_wrapper` but you're overriding higher-level defaults to None.

will this new logic sufficiently address #610?

Reply:
>  since the default is two seconds in _requests_post_rel_host

The default is 2s for generic POST requests.

>  but you're overriding higher-level defaults to None.

The default for creating predictions and groundtruths is None and is also user configurable. This is because it should scale with the number of objects included in the bulk create.

> which functions do you expect to have retry logic after this PR

The source of the bug this PR is addressing is automatic POST request retrying. I have removed the ability to auto-retry POSTing so that the user can themselves attempt to retry.

Reply:
> will this new logic sufficiently address https://github.com/Striveworks/valor/issues/610?

Yes it should, Ill update the create default to 10s by default since that was specified.

Reply:
> The source of the bug this PR is addressing is automatic POST request retrying. I have removed the ability to auto-retry POSTing so that the user can themselves attempt to retry.

does making it user-configurable give users a footgun (i.e., will they likely run into bugs if they set `timeout` manually)? I'm wondering if it makes more sense to remove the retry logic entirely, rather than turn it on for some functions and make it configurable for others

Reply:
I didnt make the retry logic configurable, retrying for POST requests is completely removed. The timeout is what is now user configurable for large bulk requests.

This makes sense because what is "bulk" could be 10 datums or 10,000 datums.

Reply:
> I didnt make the retry logic configurable, retrying for POST requests is completely removed.

take `create_groundtruths`: how is retrying for POST requests completely removed if the user can configure them with a `timeout` param?

Reply:
The user can configure the timeout for the singular request that is sent when calling `add_groundtruths`. 

This has nothing to do with retrying. If the timeout occurs or another exception is raised, the user can make the best decision for themselves on how to move forward.

Furthermore, with how variable postgres ingestion seems an ingestion of 10 datums can take less than a second or 30 seconds so there is no way to pre-bake an effective timeout. It it 100% dependent on the user's hardware.

Reply:
I'm not following what you mean here, and you didn't seem to engage with my `create_groundtruths` example; maybe we chat about it tomorrow?

Reply:
`Dataset.add_groundtruths` calls `client.add_groundtruths` so my comment above directly addresses it

> The user can configure the timeout for the singular request that is sent when calling add_groundtruths.

This has nothing to do with retrying. If the timeout occurs or another exception is raised, the user can make the best decision for themselves on how to move forward.

Reply:
I see you're now throwing a ValueError to catch any POST retries; thanks for adding that change!

I now see what you mean by some of your comments above; I was conflating the timeout logic vs. the retry logic. I'm good with these changes; thanks for adding additional tests